trustmanager/keyfilestore.go

```diff
@@ -146,8 +318,30 @@ func (s *KeyMemoryStore) ExportKey(name string) ([]byte, error) {
 
 // ImportKey imports the private key in the encrypted bytes into the keystore
 // with the given key ID and alias.
-func (s *KeyMemoryStore) ImportKey(pemBytes []byte, alias string) error {
-	return importKey(s, s.Retriever, s.cachedKeys, alias, pemBytes)
+func (s *KeyMemoryStore) ImportKey(pemBytes []byte, role, gun string) error {
+	if data.IsDelegation(role) {
```

Comment:
Similarly, `KeyMemoryStore.AddKey` checks for `keyInfo.Role == data.CanonicalRootRole || data.IsDelegation(keyInfo.Role) || !data.ValidRole(keyInfo.Role)` and sets the gun to "" - should this use the same check?